lib/etl/input/multi_influxdb.rb

```
@@ -14,17 +14,45 @@ class MultiInfluxdb < Base
     
     attr_accessor :params
 
+    # start_date : integer representing # days we gonna go back (default is 30)
+    # time_interval : symbol consisting of integer + (d, h, m) (default is 1d)
     def initialize(params, select, series, **keyword_args) 
       super()
+      @params = params
       @select = select
       @series = series
       @where = keyword_args[:where] if keyword_args.include?(:where)
       @group_by = keyword_args[:group_by] if keyword_args.include?(:group_by)
       @limit = keyword_args[:limit] if keyword_args.include?(:limit)
+      @last_stamp = keyword_args[:last_stamp] if keyword_args.include?(:last_stamp) 
+      if keyword_args.include?(:time_interval) 
+        ti = keyword_args[:time_interval].to_s 
+        if ti.end_with?("d") && ti[0..-2].match(/^(\d)+$/) 
+          @time_interval = ti[0..-2].to_i
+          @time_interval_unit = 60*60*24
+        elsif ti.end_with?("h") && ti[0..-2].match(/^(\d)+$/)
+          @time_interval = ti[0..-2].to_i
+          @time_interval_unit = 60*60
+        elsif ti.end_with?("m") && ti[0..-2].match(/^(\d)+$/)
+          @time_interval = ti[0..-2].to_i
+          @time_interval_unit = 60
+        else
+          @time_interval = 1 
+          @time_interval_unit = 60*60*24
+        end
+      else
+        @time_interval = 1 
+        @time_interval_unit = 60*60*24
+      end
+
+
+      @start_date = if keyword_args.include?(:start_date) && keyword_args[:start_date].is_a?(Integer)
```

this will silently ignore a string argument. might as well just try `to_i` on whatever the value is, and let that fail if it's not valid.